test/test_framework.py

```diff
@@ -287,6 +287,37 @@ def on_foo(self, event):
         #
         self.assertEqual(obs.seen, ["on_foo:foo=2", "on_foo:foo=2"])
 
+    def test_weak_observer(self):
+        framework = self.create_framework()
+
+        observed_events = []
+
+        class MyEvent(EventBase):
+            pass
+
+        class MyEvents(EventsBase):
+            foo = Event(MyEvent)
+
+        class MyNotifier(Object):
+            on = MyEvents()
+
+        class MyObserver(Object):
+            def on_foo(self, event):
+                observed_events.append("foo")
+
+        pub = MyNotifier(framework, "1")
+        obs = MyObserver(framework, "2")
+
+        framework.observe(pub.on.foo, obs)
+        pub.on.foo.emit()
+        self.assertEqual(observed_events, ["foo"])
+        # Now delete the observer, and note that
+        # when we emit the event, it doesn't update
+        # the local slice again
+        del obs
+        pub.on.foo.emit()
+        self.assertEqual(observed_events, ["foo"])
```

Comment:
Does that work consistently?  I was expecting we'd have to force a GC cycle here at least.

Reply:
In CPython "del object" is a refcount and deterministically removes the object.
In "Python" generically (like IronPython, Jython, etc) it is not.

If you want, we can add 'gc.collect()'. 
To be thorough, gc.collect in CPython is used to handle circular references.
 eg a.b = b; b.a = a; del a; del b;
the refcounts don't go to 0 naturally, so you need gc.collect() to find the loop and free the memory.

I'll add it just to be 'future proof', but it technically isn't necessary for existing cpython versions. (things like the attempt to do llvm, etc could change this, but none of those worked well enough to actually land.)